yt_dlp/extractor/patreon.py

```diff
@@ -185,3 +187,65 @@ def _real_extract(self, url):
                 })
 
         return info
+
+
+class PatreonUserIE(InfoExtractor):
+
+    _VALID_URL = r'https?://(?:www\.)?patreon\.com/(?P<id>[-_\w\d]+)/?(?:posts/?)?'
+
+    _TESTS = [{
+        'url': 'https://www.patreon.com/dissonancepod/',
+        'only_matching': True
+    }, {
+        'url': 'https://www.patreon.com/dissonancepod/posts',
+        'only_matching': True
+    }]
+
+    def get_entries(self, campaign_id, user_id):
+        cursor = None
+        params = {
+            'fields[campaign]': 'show_audio_post_download_links,name,url',
+            'fields[post]': 'current_user_can_view,embed,image,is_paid,post_file,published_at,patreon_url,url,post_type,thumbnail_url,title',
+            'filter[campaign_id]': campaign_id,
+            'filter[is_draft]': 'false',
+            'sort': '-published_at',
+            'json-api-version': 1.0,
+            'json-api-use-default-includes': 'false',
+        }
+
+        for page in itertools.count(1):
+
+            params.update({'page[cursor]': cursor} if cursor else {})
+            posts_json = self._download_json('https://www.patreon.com/api/posts', user_id, note='Downloading posts page [%d]' % page, query=params)
+
+            cursor = try_get(posts_json, lambda x: x['meta']['pagination']['cursors']['next'])
+            data = try_get(posts_json, lambda x: x['data'])
+
+            if data is not None:
+                for post in data:
+
+                    # self.to_screen(post)
+                    if not post['attributes']['current_user_can_view']:
+                        self.report_warning('Post not viewable by current user! Skipping!')
```

Comment:
can we atleast get the post id here?

Reply:
Yeah, we can. What would be a good way to handle it?

Reply:
If the URL is passed to PatreonIE. It will throw the relevent error, correct? In that case, just pass it along - this check isnt needed. That way, the user can get all urls if they want using `--flat-playlist`